### DISCUSSION TITLE
Conda package build

### DISCUSSION THREAD

**Discussion by Glider95** [microsoft/onnxruntime-genai#636] [Q&A]
Hello,

I would like to use this library in an environment that only allows Conda packages, not PyPi.
I have succesfully built from source however I am not sure how to create the package.
Does anyone succesfully build a Conda package and could share the configuration (meta.yml / Recipe) ? (Preferrably for Linux-64 but anything would be appreciated)
 
Best regards,

**Comment by wilderfield** [comment]
Who is setting the requirement that PyPi packages aren't allowed?

Not directly answering your question, but you can always pip install a python module into a conda environment.

You could even have a env.yaml like this to use pip to build and install the wheel from source:

```
name: customenv
channels:
  - defaults
dependencies:
  - python=3.9
  - pip
  - pip:
      - git+https://github.com/psf/requests.git@master
 ```

**Comment by Glider95** [comment]
Unfortunately, it is part of proprietary platform policy that we are using, where PiP dependencies can't be used.
Adding PiP dependencies to the Conda environment will fail the "security check".
However, we are allowed to push new Conda packages to the platform and use them.

**Reply by wilderfield** [reply]
This project looks brand new to me.

You’ll likely have to contribute your own conda recipe to conda-forge.

I see build “instructions” here
https://github.com/microsoft/onnxruntime-genai/blob/main/.github/workflows/linux-cpu-x64-build.yml

onnxruntime is in conda-forge and you could use it as a dependency of onnxruntime-genai.

I wonder though, is MSFT just going to roll this in to onnxruntime?

**Reply by Glider95** [reply]
I have tried to create my own Conda package without success.
The Cmake build is successful but not sure how to proceeed to package it after that....

**Reply by wilderfield** [reply]
CMake normally installs things into the PREFIX in a conda recipe.

if you start the conda forge PR process, I am willing to help. I’ve created one conda-forge package before.

https://conda-forge.org/docs/maintainer/adding_pkgs/
